libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt

```diff
@@ -0,0 +1,140 @@
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var loadedMapStyleFlow: MutableStateFlow<Style?>
+    private lateinit var sut: RoadNameViewBinder
+
+    @Before
+    fun setUp() {
+        ctx = ApplicationProvider.getApplicationContext()
+        navContext = spyk(
+            NavigationViewContext(
+                context = ctx,
+                lifecycleOwner = TestLifecycleOwner(),
+                viewModel = NavigationViewModel(),
+                storeProvider = { TestStore() }
+            )
+        )
+        loadedMapStyleFlow = MutableStateFlow(mockk())
+        every { navContext.mapStyleLoader } returns mockk {
+            every { loadedMapStyle } returns loadedMapStyleFlow
+        }
+
+        mapboxNavigation = mockk(relaxed = true)
+
+        sut = RoadNameViewBinder(navContext)
+    }
+
+    @Test
+    fun `bind should inflate road name layout`() {
+        val rootLayout = FrameLayout(ctx)
+
+        sut.bind(rootLayout)
+
+        assertNotNull(
+            "Expected MapboxRoadNameView",
+            rootLayout.findViewById<MapboxRoadNameView>(R.id.roadNameView)
+        )
+    }
+
+    @Test
+    fun `bind should return ReloadingComponent that attaches RoadNameComponent`() =
+        runBlockingTest {
+            val rootLayout = FrameLayout(ctx)
+            val reloadComponent = sut.bind(rootLayout) as ReloadingComponent<*>
+            reloadComponent.onAttached(mapboxNavigation)
+
+            assertTrue(reloadComponent.childComponent is RoadNameComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on Map Style change`() =
+        coroutineRule.runBlockingTest {
+            loadedMapStyleFlow.value = null
+            val rootLayout = FrameLayout(ctx)
+            val reloadComponent = sut.bind(rootLayout) as ReloadingComponent<*>
+            reloadComponent.onAttached(mapboxNavigation)
+            assertNull(reloadComponent.childComponent)
+
+            loadedMapStyleFlow.value = mockk()
+
+            assertNotNull(reloadComponent.childComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on roadNameBackground change`() =
+        coroutineRule.runBlockingTest {
+            val rootLayout = FrameLayout(ctx)
+            val reloadComponent = sut.bind(rootLayout) as ReloadingComponent<*>
+            reloadComponent.onAttached(mapboxNavigation)
+
+            val firstComponent = reloadComponent.childComponent
+            navContext.applyStyleCustomization {
+                roadNameBackground = R.drawable.mapbox_bg_button_primary
+            }
+
+            assertNotEquals(firstComponent, reloadComponent.childComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on roadNameTextAppearance change`() =
```

Comment:
We can also check that text appearance is applied.

Reply:
We could but I think we should first move that responsibility to RoadNameComponent.